src/platform.ts

```diff
@@ -70,16 +70,38 @@ export function getDefaultPowerShellPath(
     // Find the path to powershell.exe based on the current platform
     // and the user's desire to run the x86 version of PowerShell
     if (platformDetails.operatingSystem === OperatingSystem.Windows) {
-        if (use32Bit) {
-            powerShellExePath =
-                platformDetails.isOS64Bit && platformDetails.isProcess64Bit
-                    ? SysWow64PowerShellPath
-                    : System32PowerShellPath;
+        const psCoreInstallPath =
+        (!platformDetails.isProcess64Bit ? process.env.ProgramW6432 : process.env.ProgramFiles) + "\\PowerShell";
+        if (fs.existsSync(psCoreInstallPath)) {
```

Comment:
_Just_ in case someone went and created a `Program Files\PowerShell` path… we should refactor this if to say, if I’ve found a PowerShell Core, then return it, otherwise, use Windows PowerShell.

What I mean by this is:

```ts
if (fs.existsSync(psCoreInstallPath)) {

    // still do what you’re doing…

    if (psCorePaths) {
        return psCorePaths[0].exePath;
    }
}

// no ELSE, just go right to Windows PowerShell logic
if (use32Bit) {
    // …
```